### ISSUE TITLE
OMP could use a Series Index page

### ISSUE THREAD

**Issue by gemusehandler** [pkp/pkp-lib#6086]
When it comes to books (or monographs) then from an academic point of view, book series are essential. Books series provide a path to academic recognition. Books series can for instance be included in academic indexes such as Scopus, standalone books can't. Wouldn't it be a good idea when OMP allows presses to present series more prominently? We though it is! We developed a seriesindex page that provides an effect overview of the series that are included in a press. And we like to see this seriesindex page to become part of the regular OMP installation.

How did we do it? We added a catalogSeriesIndex.tpl to the installation. It requires a small additions to the index.php page and the CatelogHandler.inc.php. These modifications are highlighted in the files with: /** GEMUSEHANDLER begin modification:

Please see the files at my [GIST page at Github](https://gist.github.com/gemusehandler
)

Here is a [slightly older version](https://books.bk.tudelft.nl/index.php/press/catalog/seriesIndex
) of the page in action:

Correct: the example displays the Print/Online ISSN header when there is actually noting to display. This has been resolved in the coding on the GIST pages

**Comment by NateWr** [comment]
I'd be open to this. @asmecher and @withanage do you have any thoughts about this?

One thing I'd like to see is to have the `seriesIndex` path just be `series`. That way we have a good URL structure:

- `/catalog/series` = a list of all series
- `/catalog/series/abcd` = a list of all books in the abcd series

This would require stepping in at https://github.com/pkp/omp/blob/master/pages/catalog/CatalogHandler.inc.php#L131-L133 and passing off to the correct template.

@gemusehandler are you able to work on a PR for this (with the caveat that I'd like to hear from Alec and Dulip first)?

**Comment by withanage** [comment]
Hi @NateWr  

I think this is a very good addition and a very good  requirement.

> - `/catalog/series` = a list of all series
> - `/catalog/series/abcd` = a list of all books in the abcd series

This is  a very good path approach


One thing,  which would  be needed , if we have the series list,  is  the  [series info](https://heiup.uni-heidelberg.de/series/info/hst) page  too.

 `/catalog/series/abcd/info` or something similar.

Another idea is : If we could use the REST API for for fetching metadata, that would be very good, cause we can use the same mapping for the API and the  front-end and for performance issues, we can optimize the pagination etc. in REST in the longer run.
As we have the sortable criteria for Series in Backend, users may  want  to sort according to chronology, title or  author.
This all we can handle from the REST API.

**Comment by NateWr** [comment]
> One thing, which would be needed , if we have the series list, is the series info page too.

Shouldn't this be included in the main series page? Similar to how we include a category description at the top of a category listing or the issue description at the top of an issue table of contents.

**Comment by withanage** [comment]
> Shouldn't this be included in the main series page? Similar to how we include a category description at the top of a category listing or the issue description at the top of an issue table of contents.

yes, that is also an option.  In Heidelberg, they really needed a very customizable page as a separate landing page.
There argument was ,  I have one page, where all series are listed and from that the user have a link to a series info page without looking at the books as [here](https://heiup.uni-heidelberg.de/series) 

Another thing, that was noted is e.g. if we a large number of of books in  a series. if  we paginate, we have to have to have the header in each paginated page, which some may like, but some may not like. Considering both those, I went for the separate page in python frontend, which was also helpful when they were advertising the book series launching or wanted to refer from other sources.

**Comment by gemusehandler** [comment]
That landing page for a specific series, doesn't it already exists?
Here an example:
https://books.bk.tudelft.nl/index.php/press/catalog/series/RIUS

**Comment by NateWr** [comment]
> If we paginate, we have to have to have the header in each paginated page

Ah yeah, I'd like to solve that in the original template. Subsequent pages can show a reduced series title area. Maybe a separate issue to consider...

**Comment by asmecher** [comment]
^ As @gemusehandler points out, there's already a page for this!

**Comment by gemusehandler** [comment]
@NateWr 
I like the idea of having series instead of seriesIndex, but...
There is already a series.tpl in OMP. That is the reason I didn't use the obvious.
The series.tpl is used to display the content of a single book series.

@asmecher
The existing landing page for a series (yes, the one generated by the series.tpl) is basic but fine.
The only problem you get is when one of your series overdo it with (for instance) a very long list of scientific committee members: [CPCL](https://books.bk.tudelft.nl/index.php/press/catalog/series/CPCL)
It messes up the overview of the [seriesIndex](https://books.bk.tudelft.nl/index.php/press/catalog/seriesIndex)
I did solve it with introducing the truncate:800. It is a bit crude but it works.

**Comment by NateWr** [comment]
> There is already a series.tpl in OMP.

The template name doesn't have any impact on the URL. So at the point in the code that I linked above, you can load any template you want, like this:

```php
if (!$series) {

	// ... set up the series index template ...

	return $templateMgr->display('frontend/pages/catalogSeriesIndex.tpl');
}
```

**Comment by gemusehandler** [comment]
PR = Pull Request I believe.
Sorry guys but I am a newly here.

**Comment by NateWr** [comment]
Sorry @gemusehandler! Yep, PR = Pull Request. To provide a pull request you'll need to fork the project, get it running on your own local server based on the github repoository, and then create a branch and issue a pull request with the changes.

You should be able to find resources online describing how to create a "fork" and create pull requests. Our documentation describes [how to get OJS running on a local server](https://docs.pkp.sfu.ca/dev/documentation/en/getting-started) based off of the github repository.

**Comment by gemusehandler** [comment]
Thanks for providing some directions. I think I managed.

**Comment by asmecher** [comment]
PR: https://github.com/pkp/omp/pull/855

@gemusehandler, to keep pull requests and issues linked, we use a convention that includes the issue ID in the pull request title:
```
pkp/pkp-lib#6086 overview page for series available in a press to be added OMP
```
That'll cause the pull request to be linked to the related issue, and vice versa.

@NateWr, I see you've already been watching over this; could you take a look?

**Comment by NateWr** [comment]
Thanks @gemusehandler! I've made some comments on the PR but it is looking good. :+1: One thing we still need to figure out is how will readers get to this page? Some ideas:

- Add a link to it in the browse block on the right?
- Add a link to it in the homepage?
- Add a navigation menu item for it so it can be added to the nav menus?

Any thoughts on this? cc @withanage

**Comment by withanage** [comment]
@NateWr 
Apologies for replyling late. Somehow  missed it.
Generally the use case is that the series page has to be linked from the home page.
I prefer your third suggestion as a navigation item.
e.g. [here](https://press.nordicopenaccess.no/index.php/noasp/catalog/series/kyrkjefag) they have OMP frontend and links the individual series  under the top navigation. 

If the  top navigation can have a series index page, directly from the navigation it would be really nice.

**Comment by NateWr** [comment]
Thanks @withanage! @gemusehandler did you have any questions about the comments here or on the pull request? We can talk through anything you need to continue work on it.

**Comment by gemusehandler** [comment]
I think I managed.

**Comment by rinorazzi** [comment]
Hi @NateWr @gemusehandler
I'm very interested in this new feature you are wonking on.
I tried to apply it on a fresh installation of OMP 3.3.0-4 where the Default OMP theme is used. 
I replaced in my OMP the 3 file your Pull Request : pkp/omp#855
but I was not able to make it run.
When i try to view the Series Index page with 
/index.php/jip/catalog/seriesIndex
 I get an "error 500".
Is this Pull Request compatible with OJS 3.3.0?
If yes, could you please help me in getting it work on my website?
